contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,559 @@
+the plug-in chosen. Recycling will not be a supported reclaim policy. Once the user deletes the claim against a PV, the volume will 
+be scrubbed according to how it was bound. The path value in the local PV definition would be overloaded to define the path of the raw
+block device rather than the fileystem path.
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeType: block #proposed API change
+capacity:
+storage: 100Gi
+local:
+  path: /dev/xvdc 
+accessModes:
+  - ReadWriteOnce
+persistentVolumeReclaimPolicy: Delete 
+```
+
+## Storage Class API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeType
+as 'block' but the fsType as 'xfs'.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: kubernetes.io/local-block-glusterfs
+parameters:
+  volumeType: block #opaque value  / plug-in dependent
+  fsType: xfs
+```
+The provisioner (if applicable) should validate the parameters and return and error if the combination specified is not supported.
+This also allows the use case for leveraging a Storage Class for utilizing pre-defined static volumes. By labeling the Persistent Volumes
+with the Storage Class, volumes can be grouped and used according to how they are defined in the class.
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+name: block-volume
+provisioner: no-provisioning 
+parameters:
+```
+  
+# Use Cases
+
+## UC1: 
+
+DESCRIPTION: An admin wishes to pre-create a series of raw block devices to expose as PVs for consumption. The admin wishes to specify the purpose of these devices by specifying 'block' as the volumeType for the PVs.
+
+WORKFLOW:
+
+ADMIN:
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local-raw-pv
+spec:
+  volumeType: block
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete
+```
+
+## UC2: 
+
+DESCRIPTION: 
+
+A user uses a container on virtual machine on hypervisor such as KVM, VMware and wishes to use raw block device for databases such as MariaDB. 
+
+BACKGROUND:
+
+Using hypervisor's device passthrough feature(KVM: device passthrough, VMware: RawDeviceMapping), storage admin can expose raw block device into virtual machine then user can consume it via PV and PVC. Also dynamic provisioning could work if external-provisioner support this.
+
+WORKFLOW:
+
+ADMIN:
+
+* Admin creates a disk and attach it to server and pass-through the disk to KVM guest which Kubelet node1 is working inside.
```

Comment:
@jsafrane 
IIRC, following steps are needed to use pre-created FC volume.(without external provisioner)
- Admin create a volume and expose the volume to a worker node.(This is done by storage operation)
- Admin create FC PV using device location such as WWN and LUN.
- User creates a PVC to bind pre-created FC PV.
- User creates a Pod with PVC. During Pod creation, FC Plugin find FC disk which is attached to the worker node using "echo 1 > /sys...", then the FC volume is recognized to the worker node and the Pod consume this FC volume. 

As you mentioned, I think FC volume is similar to a HostPath without external provisoner. But I think this is not a local volume case, because the FC volume is network attached volume and is not included inside the server local.